### PR TITLE
New APIs for container instance

### PR DESCRIPTION
**Pull request by yangl900**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ X ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ X ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ X ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ X ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by yangl900** [comment]
I believe we need WaitForARMFeedback tag.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-10-01/containerInstance.json [hunk 1]
```diff
@@ -544,6 +544,68 @@
...
+          },
+          {
+            "$ref": "#/parameters/LocationParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
```
Comment (ravbhatnagar) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-10-01/containerInstance.json]: Please add the "default" response to catch all the error states. Sample in batch RP swagger.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-10-01/containerInstance.json [hunk 2]
```diff
@@ -1585,6 +1651,111 @@
...
+          },
+          "description": "The list of cached images."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URI to fetch the next page of cached images."
+        }
+      }
+    },
+    "cachedImages": {
```
Comment (ravbhatnagar) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-10-01/containerInstance.json]: does it make sense for these to have the ARM resource Id?

Comment (ravbhatnagar) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-10-01/containerInstance.json]: Is there a future scenario for these to be shared and reused? Then what happens to RBAC?

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-10-01/containerInstance.json [hunk 3]
```diff
@@ -544,6 +544,68 @@
...
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/CachedImagesListResult"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.ContainerInstance/locations/{location}/capabilities": {
```
Comment (ravbhatnagar) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-10-01/containerInstance.json]: Do you have skus for your resource types? likely not, but seems like convergence will make it easier for customers/client to discover the capabilities for a resource type.

Comment (ravbhatnagar) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-10-01/containerInstance.json]: If not, this should be fine, other RPs also have similar APIs. At some point, ARM will need to provide a converged experience for such scenarios across RPs.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-10-01/containerInstance.json (hunk 2 above)
Comment (yangl900) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-10-01/containerInstance.json]: it's right now a public, relatively static list. Anyone have permission to the subscription should be able to read the list.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-10-01/containerInstance.json (hunk 3 above)
Comment (yangl900) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-10-01/containerInstance.json]: there is no SKUs for the resource type. It's a flexible combination of CPU and memory, thus only have limit, but not a fixed set of SKUs.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-10-01/containerInstance.json (hunk 2 above)
Comment (yangl900) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-10-01/containerInstance.json]: I'd say it's not necessary right now, but if really needed, adding an Id is not a problem.

Comment (ravbhatnagar) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-10-01/containerInstance.json]: ok, will sign off on the manifest. lets add the id.

**Review by yangl900 — COMMENTED** [review]
Resolved the comments. @ravbhatnagar please help take a look again.

**Review by KrisBash — APPROVED** [review]
LGTM

**Comment by yangl900** [comment]
anyone can help merge this PR? @jianghaolu ?

**Comment by ravbhatnagar** [comment]
Re-Signing off from ARM side.

**Review by jianghaolu — APPROVED** [review]

**Pull request merged**